src/libxively/sft/xi_sft_logic.c

```diff
@@ -62,37 +68,178 @@ xi_state_t xi_sft_on_connected( xi_sft_context_t* context )
     }
 
     xi_control_message_t* message_file_info = xi_control_message_create_file_info(
-        context->updateable_files, NULL, context->updateable_files_count );
+        context->updateable_files, NULL /* todo_atigyi: handle revisions */,
+        context->updateable_files_count );
 
     ( *context->fn_send_message )( context->send_message_user_data, message_file_info );
 
     return XI_STATE_OK;
 }
 
 xi_state_t
-xi_sft_on_message( xi_sft_context_t* context, const xi_control_message_t* sft_message )
+xi_sft_on_message( xi_sft_context_t* context, xi_control_message_t* sft_message_in )
 {
-    if ( NULL == context || NULL == sft_message )
+    if ( NULL == context || NULL == sft_message_in )
     {
         return XI_INVALID_PARAMETER;
     }
 
-    switch ( sft_message->common.msgtype )
+    xi_state_t state = XI_STATE_OK;
+
+    switch ( sft_message_in->common.msgtype )
     {
-        case XI_CONTROL_MESSAGE_SC_FILE_UPDATE_AVAILABLE:
-            /* - check whether device is ready to start download of file
-               - get the first chunk of file in question with FILE_GET_CHUNK */
-            break;
-        case XI_CONTROL_MESSAGE_SC_FILE_CHUNK:
+        case XI_CONTROL_MESSAGE_SC__SFT_FILE_UPDATE_AVAILABLE:
+        {
+            /* todo?: check whether device is ready to start download of file */
```

Comment:
yeah, seems like a good idea.  We should think about how that might be done for version 1.1